### PR TITLE
Enable MKL-DNN in Naive Executor and Analysis Predictor

### PR DESCRIPTION
**Pull request by Sand3r-**
Enables MKL-DNN in AnalysisPredictor so that the accuracy of pass-fused layers can be assessed in tests such as `test_analyzer_ocr`.

### CODE REVIEW

**Review by sfraczek — APPROVED** [review]
LGTM!

**Comment by sfraczek** [comment]
It doesn't work when built without mkldnn (`-DWITH_MKLDNN=OFF`)

File: paddle/fluid/framework/naive_executor.cc [hunk 1]
```diff
@@ -146,5 +146,22 @@ void NaiveExecutor::CleanFeedFetchOps() {
   ops_.swap(ops);
 }
 
+void NaiveExecutor::EnableMKLDNN(const ProgramDesc &program) {
+#ifdef PADDLE_WITH_MKLDNN
+  VLOG(3) << "use_mkldnn=True";
+  for (size_t block_id = 0; block_id < program.Size(); ++block_id) {
+    auto *block = const_cast<ProgramDesc &>(program).MutableBlock(block_id);
+    for (auto *op : block->AllOps()) {
+      if (op->HasAttr("use_mkldnn")) {
```
Comment (hshen14) [inline paddle/fluid/framework/naive_executor.cc:155]: Where is op flag use_mkldnn false? Should it be always true?

File: paddle/fluid/inference/tests/api/analyzer_vis_tester.cc [hunk 2]
```diff
@@ -61,8 +61,6 @@ void SetConfig(AnalysisConfig *cfg) {
   cfg->ir_passes.push_back("fc_gru_fuse_pass");
 #ifdef PADDLE_WITH_MKLDNN
   cfg->_use_mkldnn = true;
-  // disable mkldnn fuse since it should have some bugs
```
Comment (hshen14) [inline paddle/fluid/inference/tests/api/analyzer_vis_tester.cc:64]: Will you add back?

File: paddle/fluid/framework/naive_executor.cc (hunk 1 above)
Comment (Sand3r-) [reply paddle/fluid/framework/naive_executor.cc:155]: It is used to determine whether the MKL-DNN version of the layer should be ran for a given operation.

File: paddle/fluid/inference/tests/api/analyzer_vis_tester.cc (hunk 2 above)
Comment (Sand3r-) [reply paddle/fluid/inference/tests/api/analyzer_vis_tester.cc:64]: Now that the MKL-DNN is properly called the test no longer fails when this pass is used. There is no need for disabling it.

**Review by luotao1 — APPROVED** [review]
LGTM！Thanks very much before we implement EnableMKLDNNPass.

**Pull request merged**